overlord/hookstate/ctlcmd/set.go

```diff
@@ -42,6 +56,51 @@ func (s *setCommand) Execute(args []string) error {
 		return fmt.Errorf("cannot set without a context")
 	}
 
-	// TODO: Talk to the handler to take care of the set request.
+	// Extract the transaction from the context. If none, make one.
+	s.context().Lock()
+	transaction, ok := s.context().Cached("transaction").(*configstate.Transaction)
+	s.context().Unlock()
+	if !ok {
+		var err error
+		transaction, err = initializeTransaction(s.context())
+		if err != nil {
+			return fmt.Errorf("cannot initialize transaction: %s", err)
+		}
+	}
+
+	for _, patchValue := range s.Positional.ConfValues {
+		parts := strings.SplitN(patchValue, "=", 2)
+		if len(parts) != 2 {
+			return fmt.Errorf(i18n.G("invalid parameter: %q (want key=value)"), patchValue)
+		}
+		key := parts[0]
+		var value interface{}
+		err := json.Unmarshal([]byte(parts[1]), &value)
+		if err != nil {
+			// Not valid JSON-- just save the string as-is.
+			value = parts[1]
+		}
+
+		transaction.Set(s.context().SnapName(), key, value)
+	}
+
+	s.context().Lock()
+	s.context().Cache("transaction", transaction)
```

Comment:
This needs to be done inside the initial lock, and not every time. Being an in-memory value, we don't need to set it back after we obtain it. When we Set we are mutating that in-memory value already.